src/arrayMethodSort.js

```diff
@@ -3,9 +3,22 @@
 /**
  * Implement method Sort
  */
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(cb) {
+    for (let i = 0; i < this.length - 1; i++) {
+      const currElem = this[i];
+      const nextElem = this[i + 1];
+
+      if (cb ? cb(nextElem, currElem, i, this) < 0
+        : nextElem.toString()[0] < currElem.toString()[0]) {
+        this.splice(i + 1, 1);
```

Comment:
The use of splice and unshift in the loop can have a high computational complexity. Consider finding a more efficient way to sort the array.